### DISCUSSION TITLE
Neo6 GPS with Tx and Rx?

### DISCUSSION THREAD

**Discussion by F4u5t** [witnessmenow/ESP32-Cheap-Yellow-Display#75] [Q&A] [answered]
Got a new project idea, and im wondering if i can use the P1 pins ( Vin, Tx,Rx,Gnd) to talk to a hardware serial GPS module, like This one?
how would i define and wire the pins??

https://www.amazon.com/HiLetgo-GY-NEO6MV2-Controller-Ceramic-Antenna/dp/B01D1D0F5M/ref=sr_1_1?crid=1JTFFJO99Y887&keywords=neo6mv2&qid=1701194538&s=electronics&sprefix=neo6mv2%2Celectronics%2C83&sr=1-1

**Comment by cedarlakeinstruments** [comment, marked as answer]
I used a similar GPS (before abandoning the idea to use this display and switching to an STM32 instead). I think I had it on P1:

const int RX_PIN = 17;
const int TX_PIN = 18;

void setup() 
{
    setupUI();

    Serial2.begin(9600, SERIAL_8N1, RX_PIN, TX_PIN);
}

**Reply by F4u5t** [reply]
awesome, I ll test that out!

**Reply by F4u5t** [reply]
how did you figure out pins 17 and 18 - the documentation SUCKS!

**Reply by cedarlakeinstruments** [reply]
I agree!
Don't remember how I figured it out: I spent a lot of time reading whatever scraps of documentation I could find.

**Reply by samurairai** [reply]
i got Cheap Yellow Display use the same pin too but failed to do simple loop back of Pin 17 to 18.  However, I can do the same on other ESP32.  Any idea what was the problem?

**Reply by lohikarhu** [reply]
I'm working with stm32f411 and f303...i wondered if you have published anything on your GPS interface; I'm planning on buying a neo8 module to do some on-car 'stuff'  
keep well

**Comment by F4u5t** [comment]
I got it talking with pins 1 and 3, but that also seems t9 be the usb port
I think usb and port p1 are connected

**Comment by Jmac8421** [comment]
I got a Neo-6M GPS module working on CYD by soldering 2 wires to the RGB LED (IO17 & IO16). The RGB LED can be left on the board but will blink Green.  TinyGPS++.h can be used with Serial2.begin(9600), no serialsoft. For those who are lazy just one wire can  be used from TX on the Neo-6M to the pin closest to WROOM-32 on the RCB.

#include <TFT_eSPI.h>
#include <TinyGPS++.h>

#define GPS_BAUDRATE 9600  // The default baudrate of NEO-6M is 9600

TinyGPSPlus gps;  // the TinyGPS++ object
TFT_eSPI tft = TFT_eSPI();

void setup() {
  Serial.begin(9600);
  Serial2.begin(GPS_BAUDRATE);  // IO: 17 TX2, 16 RX2 (pin 28, 27)
 Serial.println(F("ESP32 - GPS module"));
 tft.init();
  tft.setRotation(1);  //This is the display in landscape

**Comment by isyed757** [comment]
can I use the io pins on the cyd with the neo 6m gps? (I am new to electronics)

**Comment by stevedee78** [comment]
https://github.com/stevedee78/CYD_GPS_Speedometer
